connectors/byoc.py

```diff
@@ -668,71 +616,101 @@ async def prepare_docs(self, data_provider, pipeline=None, filtering=None):
 
     async def prepare(self, config):
         """Prepares the connector, given a configuration
-
-        If the connector id and the service is in the config, we want to
+        If the connector id and the service type is in the config, we want to
         populate the service type and then sets the default configuration.
-
-        Returns the source class.
         """
         configured_connector_id = config.get("connector_id", "")
-        configured_service_type = config.get("service_type", "")
+        if self.id != configured_connector_id:
+            return
+
+        if self.service_type is not None and not self.configuration.is_empty():
+            return
```

Comment:
Both ifs should have some debug logging to tell that something went wrong. Maybe first one should actually be a warning.

Reply:
This `prepare` method is for `Build a connector` connector, and we expect it always returns here after its first execution.

During the first execution, this method will update the `service_type` (from config) and the `configuration` (from connector implementation) for the `Build a connector `connector.

IMO, we should only log the first execution, and silently return for the rest.